pkg/agent/openflow/pipeline.go

```diff
@@ -41,38 +41,141 @@ import (
+	// - Decide where to add the new table in the pipeline. The order table declaration decides the order of tables in the
+	//   stage. For example:
+	//     * If you want to add a table called `FooTable` between `SpoofGuardTable` and `IPv6Table` in pipelineIP, then
+	//       the table should be declared after `SpoofGuardTable` and before `IPv6Table`:
+	//       ```go
+	//          SpoofGuardTable  = newTable("SpoofGuard", stageValidation, pipelineIP)
+	//          FooTable         = newTable("Foo", stageValidation, pipelineIP)
+	//          IPv6Table        = newTable("IPv6", stageValidation, pipelineIP)
+	//       ```
+	//      * If you want to add a table called `FooTable` just before `ARPResponderTable` in pipelineARP, then the table
+	//        should be declared before `ARPResponderTable`:
+	//       ```go
+	//          FooTable          = newTable("Foo", stageOutput, binding.PipelineARP)
+	//          ARPResponderTable = newTable("ARPResponder", stageOutput, binding.PipelineARP)
+	//       ```
+	//       * If you want to add a table called `FooTable` just after `ConntrackStateTable` in pipelineARP, then the
+	//         table should be declared after `ConntrackStateTable`:
+	//       ```go
+	//          SNATConntrackTable  = newTable("SNATConntrackZone", stageConntrackState, pipelineIP)
+	//          ConntrackTable      = newTable("ConntrackZone", stageConntrackState, pipelineIP)
+	//          ConntrackStateTable = newTable("ConntrackState", stageConntrackState, pipelineIP)
+	//          FooTable            = newTable("Foo", stageConntrackState, pipelineIP)
+	//       ```
+	//  - Reference the new table in a feature in file pkg/agent/openflow/framework.go. The table can be referenced by multiple
+	//    features if multiple features need to install flows in the table. Note that, if the newly added table is not
+	//    referenced by any feature or the features referencing the table are all inactivated, then the table will not
+	//    be realized in OVS; if at least one feature referencing the table is activated, then the table will be realized
+	//    at the desired position in OVS pipeline.
+	//
+	// How to forward packet between tables with a proper action in FlexiblePipeline?
+	//
+	// |   table A   | |   table B   | |   table C   | |   table D   | |   table E   | |   table F   | |   table G   |
+	// |   stage S1  | |                           stage S2                          | |          stage S4           |
+	//
+	//  - Next is used to forward packets to the next table. E.g, A -> B, B -> C, C -> D, etc.
+	//  - GotoTable is used to forward packets to a specific table, and the target table ID should be greater than the
+	//    current table ID. Note that, it is recommended to use GotoTable to forward packets with a stage, e.g, B -> D,
+	//    C -> E. However, in future OVS pipeline development, if packets need to be forwarded to a table in another stage
```

Comment:
Today we do not have the case, but if in future there is a case that a packet needs to be forwarded to a table in another stage directly, GotoTable can also be used, e.g.